[Windows] Added a test for _getFilesystemRepresentation

gmittert opened the pull request:
A small test to ensure that Windows FSRs don't regress

compnerd commented:
@gmittert would you mind rebasing this change?  I think most of the issues are taken care of, but I think it would get good ensure that everything is merged before closing this out.

gmittert commented:
@compnerd looks like you got all the those I found. The only thing left is the test that I added. I've rebased the change to just be the added test now.

gmittert commented:
@swift-ci please test windows

gmittert commented:
@swift-ci please test

compnerd commented:
@swift-ci please test Linux platform

compnerd commented:
Windows CI: https://dev.azure.com/compnerd/swift-build/_build/results?buildId=30730&view=results

lxbndr commented on Tests/Foundation/Tests/TestFileManager.swift:
```diff
@@ -1755,6 +1755,55 @@ VIDEOS=StopgapVideos
+            "\(tmpPath)\\testfile",       // C:/Users/...\testfile
+            "\(tmpPath)/testfile",        // C:/Users.../testfile
+            "\(tmpPath)/testfile",        // /Users/user/.../testfile
+            "\(tmpPath.first!):testfile", // C:testfile
```
Btw, this fails when runner has no Administrator privileges. CI agents are usually elevated, but manual run as user would report failure.
